settings/src/script.js

```diff
@@ -130,6 +130,15 @@ function Main( { userId } ) {
 		return <Spinner />;
 	}
 
+	const { '2fa_available_providers': availableProviders, '2fa_revalidation': revalidation } =
+		record;
+	// Check that there are providers, and that there isn't only backup codes.
+	const hasPrimaryProvider =
+		!! availableProviders.length &&
+		! (
+			availableProviders.length === 1 && availableProviders[ 0 ] === 'Two_Factor_Backup_Codes'
+		);
+
```

Comment:
I'm wondering if it would be better to either a) have an array of "not valid" providers OR b) a list of primary providers.

Eg
```suggestion
	// Check that there are expected providers enabled.
	const primaryProviders = [ 'Two_Factor_Totp', 'TwoFactor_Provider_WebAuthn' ];
	const hasPrimaryProvider =
		!! availableProviders.filter( provider => primaryProviders.includes( provider ) ).length;

```

Reply:
Otherwise, ignoring certain providers is probably also viable, but for testing on sandboxes, I'd recommend explicitly ignoring the Dummy method too, even though that shouldn't be seen in production.

```suggestion
	// Check that there are expected providers enabled.
	const ignoredProviders = [ 'Two_Factor_Backup_Codes', 'Two_Factor_Dummy' ];
	const hasPrimaryProvider =
		!! availableProviders.filter( provider => ! ignoredProviders.includes( provider ) ).length;

```

Reply:
Yeah, just noticed @iandunn's [suggestion](https://github.com/WordPress/wporg-two-factor/pull/161#issuecomment-1546009919) about reusing the `primaryProviderEnabled` derived value from https://github.com/WordPress/wporg-two-factor/blob/trunk/settings/src/components/account-status.js#L25

Having centralized logic for this probably makes sense.

Reply:
Ahh yup, I didn't see that either. That makes sense to me :)

Reply:
Updated to use a new boolean on `userRecord`